src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -0,0 +1,232 @@
+		return engineConfig.targetPatterns.concat(targetPatterns);
+	}
+
+	filterUnsupportedPaths(paths: string[]): string[] {
+		const filteredPaths = paths.filter(p => p.endsWith(".ts"));
+		this.logger.trace(`Input paths: ${paths}, Filtered paths: ${filteredPaths}`);
+		return filteredPaths;
+	}
+
+	async getRunConfig(target?: string): Promise<Record<string, any>> {
+		const tsconfigPath = await this.findTsconfig(target);
+
+		if (!tsconfigPath) {
+			throw new Error(`Unable to find ${TS_CONFIG}. Please provide ${TS_CONFIG} in the target ${target} or the current working directory or in Config.`);
+		}
+
+		// Alert the user we found a config file, if --verbose
+		this.alertUser(tsconfigPath);
+
+		const config = {};
+
+		Object.assign(config, ES_PLUS_TS_CONFIG);
+
+		// Enable typescript by registering its project config file
+		config["parserOptions"].project = tsconfigPath;
+
+		this.logger.trace(`Using config for run: ${JSON.stringify(config)}`);
+
+		return config;
+	}
+
+	private alertUser(tsconfigPath: string): void {
+		const event = {
+			messageKey: 'info.usingEngineConfigFile', args: [tsconfigPath], type: 'INFO', handler: 'UX', verbose: true, time: Date.now()
+		};
+		this.outputProcessor.emitEvents([event]);
+	}
+
+	async findTsconfig(target: string): Promise<string> {
+		let tsconfigPath: string;
+
+		// Check config
+		tsconfigPath = await this.getTsconfigFromConfig();
+
+		// Check target
+		if (!tsconfigPath) {
+			tsconfigPath = await this.getTsconfigFromTarget(target);
+		}
+
+		// Check current working directory
+		if (!tsconfigPath) {
+			tsconfigPath = await this.checkDirectoryForTsconfig('');
+		}
+
+		this.logger.trace(`Using ${TS_CONFIG} from ${tsconfigPath}`);
+		return tsconfigPath;
+	}
+
+	private async getTsconfigFromConfig(): Promise<string> {
+		let tsconfigPath = '';
+		// Check if overriddenConfig is available
+		const overriddenConfig = this.config.getOverriddenConfigPath(this.getName());
+		if (!overriddenConfig) {
+			this.logger.trace(`Did not find an overridden path from Config`);
+			return tsconfigPath;
+		}
+
+		// Complain if overriddenConfig does not exist.
+		if (!(await this.fileHandler.exists(path.resolve(overriddenConfig)))) {
+			throw new SfdxError(`Invalid path in Config: ${overriddenConfig}`); // FIXME: move to messages.js
+		}
+
+		// If overriddenConfig is a directory, look for tsconfig.json file.
+		if (await this.fileHandler.isDir(overriddenConfig)) {
+			this.logger.trace(`Looking for ${TS_CONFIG} inside overridden dir from config ${overriddenConfig}`);
+			tsconfigPath = await this.checkDirectoryForTsconfig(overriddenConfig);
+			// Complain if we can't find a tsconfig.json inside.
+			if (!tsconfigPath) {
+				throw new SfdxError(`Overridden path [${overriddenConfig}] does not contain ${TS_CONFIG}`);
+			}
+		} else if (overriddenConfig.endsWith(TS_CONFIG)) { // If overriddenConfig is a file, check if the file is tsconfig.json
```

Comment:
I think this could fail if the user creates a file named 'my-tsconfig.json' and uses that value in the configuration override? Maybe use https://nodejs.org/api/path.html#path_path_parse_path to get the file name.

Reply:
Good catch. Fixing this.

Reply:
Apparently, to accept a config file name other than `tsconfig.json`, typescript compiler requires a `-p` option - I'm not sure if we need to do anything more to get eslint to accept config with a different filename.
https://github.com/microsoft/TypeScript/issues/38983

Will leave this logic as such until we discover how eslint can use a different config filename. And also a Todo.